From 7cb519dd4859a799a0fbfaeaab8bf1f8d0c13158 Mon Sep 17 00:00:00 2001
From: Kalesh Singh <kaleshsingh@google.com>
Date: Mon, 6 Dec 2021 16:19:58 -0800
Subject: [PATCH] Allow shell to read tracefs entries

Bug: 209513178
Test: adb shell ls -la /sys/kernel/tracing/events
Test: simpleperf record -e cpu-clock --trace-offcpu sleep 1
Change-Id: I554d6a47a5d55e52f2470cdebb5017c12a019ff4
---
 daemon/main.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/daemon/main.cpp b/daemon/main.cpp
index d97e8f9d..1efa797c 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -111,10 +111,11 @@ static void drop_privileges(int server_port) {
     // AID_UHID for using 'hid' command to read/write to /dev/uhid
     // AID_EXT_DATA_RW for writing to /sdcard/Android/data (devices without sdcardfs)
     // AID_EXT_OBB_RW for writing to /sdcard/Android/obb (devices without sdcardfs)
+    // AID_READTRACEFS for reading tracefs entries
     gid_t groups[] = {AID_ADB,          AID_LOG,          AID_INPUT,    AID_INET,
                       AID_NET_BT,       AID_NET_BT_ADMIN, AID_SDCARD_R, AID_SDCARD_RW,
                       AID_NET_BW_STATS, AID_READPROC,     AID_UHID,     AID_EXT_DATA_RW,
-                      AID_EXT_OBB_RW};
+                      AID_EXT_OBB_RW,   AID_READTRACEFS};
     minijail_set_supplementary_gids(jail.get(), arraysize(groups), groups);
 
     // Don't listen on a port (default 5037) if running in secure mode.
-- 
2.39.0

